### PR TITLE
refactor and add option to mask output

### PR DESCRIPTION
**Pull request by raphaeldussin**
@jkrasting this could help with https://github.com/raphaeldussin/om4labs/pull/69

it implements a masking option for the streamfunction so we can remove that step from om4labs

### CODE REVIEW

**Comment by raphaeldussin** [comment]
still some hardcoded stuff to fix though

File: xoverturning/compfunc.py [hunk 1]
```diff
@@ -165,8 +172,24 @@ def select_basins(
...
+        if "yq" in maskbasin.dims:
+            grid = define_grid(ds)
+            bathy_coloc = grid.interp(ds[bathy], "Y", boundary="fill")
+        elif "yh" in maskbasin.dims:
+            bathy_coloc = ds[bathy]
+        else:
+            raise ValueError("Unsupported coord")
+        bathy_basin = bathy_coloc.where(maskbasin).fillna(0.0)
+        max_depth = bathy_basin.max(dim="xh")
+        maskmoc = xr.where(ds["z_i"] > max_depth, 0, 1)
```
Comment (jkrasting) [inline xoverturning/compfunc.py]: "z_i" will not be present when using the "rho_2" vertical coord. and there is no check here to determine the coordinate system

**Review by jkrasting — CHANGES_REQUESTED** [review]

File: xoverturning/compfunc.py (hunk 1 above)
Comment (raphaeldussin) [reply xoverturning/compfunc.py]: yes, I have a bit of cleanup to do

**Comment by raphaeldussin** [comment]
@jkrasting do you want to test this again with your om4labs moc diag? it should work for rho2 as well now

File: xoverturning/moc.py [hunk 2]
```diff
@@ -58,33 +55,98 @@ def calcmoc(
...
-        u_ctr, v_ctr = rotate_velocities_to_geo(ds, ucorr, vcorr)
+        u_ctr, v_ctr = rotate_velocities_to_geo(ds, ucorr, vcorr, names)
     else:
-        u_ctr, v_ctr = interp_to_grid_center(ds, ucorr, vcorr)
+        u_ctr, v_ctr = ucorr, vcorr
+
+    # check vertical dimensions are in the dataarray
+    layer = names["layer"]
+    if layer not in v_ctr.dims:
+        raise ValueError(f"{layer} not found in transport array")
```
Comment (jkrasting) [inline xoverturning/moc.py:76]: This line still gets triggered somehow when I run with `rho2` output.

**Review by jkrasting — CHANGES_REQUESTED** [review]

File: xoverturning/moc.py (hunk 2 above)
Comment (jkrasting) [reply xoverturning/moc.py:76]:
``` 
File "/Users/krasting/pkgs/xoverturning/xoverturning/moc.py", line 76, in calcmoc
    raise ValueError(f"{layer} not found in transport array")
ValueError: z_l not found in transport array
```

Comment (jkrasting) [reply xoverturning/moc.py:76]: Disregard ... I forgot to set the `vertical` kwarg.

**Review by jkrasting — APPROVED** [review]

File: xoverturning/moc.py (hunk 2 above)
Comment (raphaeldussin) [reply xoverturning/moc.py:76]: it made me realize the readme example was broken (bad copy/paste)

**Pull request merged**